src/service/file.svc.ts

```diff
@@ -57,49 +113,23 @@ export function validateDirectory(dirPath: string): void {
 /**
  * Saves an SBOM to a file in the specified directory
  */
-export function saveSbomToFile(dir: string, sbom: CdxBom): string {
-  const outputPath = join(dir, `${filenamePrefix}.sbom.json`);
-
-  try {
-    fs.writeFileSync(outputPath, JSON.stringify(sbom, null, 2));
-    return outputPath;
-  } catch (error) {
-    throw new Error(`Failed to save SBOM: ${getErrorMessage(error)}`);
-  }
+export function saveSbomToFile(dir: string, sbom: CdxBom, outputPath?: string): string {
+  const { fileName, fullPath } = resolveOutputPath(dir, `${filenamePrefix}.sbom.json`, outputPath);
+  return writeJsonFile(fullPath, fileName, sbom, fileName);
 }
 
 /**
  * Saves a trimmed SBOM to a file in the specified directory
  */
 export function saveTrimmedSbomToFile(dir: string, sbom: CdxBom): string {
```

Comment:
this looks very similar to `saveReportToFile ` not sure if it make sense to keep 2 specialized function or make 1 generic, feel free to ignore